### DISCUSSION TITLE
Starlink Anyone?

### DISCUSSION THREAD

**Discussion by docderwood** [Ysurac/openmptcprouter#1756] [Q&A]
Any of you using Starlink? I’m hoping to bond two starlink connections with LTE backup

**Comment by Ysurac** [comment]
I don't have starlink (not yet available in France), but some users reported Starlink is working.
I don't know what/where is the connection limit with starlink, but if both can work at full speed at the same time I don't see why this wouldn't work.

**Comment by bdruth** [comment]
I have Starlink working, bonded to a DSL connection. No IPv6 on the Starlink connection, but otherwise works well. Using vpsserver.com currently, since they have a DC in Chicago.

**Comment by Adorfer** [comment]
see ticket https://github.com/Ysurac/openmptcprouter/issues/1767

**Comment by docderwood** [comment]
FYI, I have this working well with the exception of a system I use that requires Cisco Anyconnect VPN. It will connect, but unusable unfortunately.

**Reply by Ysurac** [reply]
Difficult for me to test Cisco Anyconnect VPN as I don't have it. You can try to customize MTU.
If it use UDP, you can try also to use V2Ray as proxy in advanced features of Wizard page.

**Comment by kevin39** [comment]
Hi,
FYI, I'm the owner of a Starlink connection since 10 days ago (France). It works great.
OMR is working great, no issue related to my Starlink connection.

Total bandwidth : 300mbs (280mbs from Starlink + ~10mbs 4G + ~1mbs DSL) using my Rpi4 and MilkyWay VPS.

edit : Starlink App is working too (need a static route to 192.168.100.1)

**Reply by bjoerrrrn** [reply]
Can you give us your MPTCP Settings (Network → MPTCP) please?

**Reply by kevin39** [reply]
Hi, 
What's your issue ?
![image](https://user-images.githubusercontent.com/976058/137698787-08eb28b2-9f44-46d3-b2ba-6bceba17301a.png)

**Reply by bjoerrrrn** [reply]
I can get full bw on every single gateway with omr-test-speed (interface).
With bonding the theoretical limit could be 250+16+~170 from Starlink around 436 MBit/s in Download and around 80MBit/s in Upstream. I can reach around 220 MBit/s Down and around 68 MBit/s up. Upload is fine tho. 

_EDIT_ With your Settings, i reach 312/72. Lovely! Thanks! _/EDIT_

**Comment by digitalsolutiontop** [comment]
hi, I have starlink in italy (200/30 average 350/50 Mbit peaks) + 4g (50 Mbit / 40 Mbit) with omr on rasp4. ( these days i am testing linksys wrt3200acm )

in aggregation I dont have speed benefits at the moment, but I think it is a problem of the current routing by starlink
on the other hand I have the public ip and this is enough for me at the moment

i have 3 vps with 3 different location and ip for test

1 france ovh (it is the one that performs best )
1 frosinone seeweb, near rome
1 frosinone serverplan, near rome

the two italian datacenters are the ones closest to starklink pop in rome

the latter 2 do not exceed a total download speed of 150 Mbit

**Reply by kamenp** [reply]
Any luck with the linksys wrt3200acm? I just picked up a used one and installed OMR on it with the intent to bond starlink, 4g and wifi.

**Reply by iambenmitchell** [reply]
> wifi

What do you mean by bonding with WiFi?

People have had success with starlink, not sure about the wrt3200acm though. Personally I use a Raspberry pi 4 with a smart switch (VLANs)

**Comment by redkite127** [comment]
Hi there, I'm using two Starlink (among 4 WANs). Aggregation performs well, I'm also using a OVH VPS endpoint :+1: 
I still keep the ADSL as master (10Mbps, but reliable).

**Comment by Chris5342** [comment]
Anyone got Starlink V2 going? (Square Ariel)

I have tried turning on bypass mode but it's not clear what the IP and Gateway should be.

I see many above have got it going but not much on exactly how.

**Reply by bdruth** [reply]
I don't have the square dish, I have the original round dish. I'm not using the Starlink router, just Ethernet straight to OMR box. Bypass should work the same? What issues are you having?

**Reply by Chris5342** [reply]
Yes I think bypass should be the same.
I'm not sure what to put in these boxes and wondered what you did:
![Screenshot 2022-05-12 at 20 44 55](https://user-images.githubusercontent.com/105393840/168190207-9714d09d-14a4-4c69-a316-71d53c73718e.jpg)

Googling seems to say that Dishy will hand out an IP but this will be a CGNAT 100.*.*.* IP.

But it's not clear that OMR would get that automatically and we need to put something in the boxes anyway!
Also it seems likely the IP would change.

I was going to try using the 100.*.*.* IP but was hoping to get guidance from the all the folks here who say they got Starlink going but not how :) ie @redkite127 @digitalsolutiontop @bjoerrrrn @kevin39 
Thank you.

**Reply by Chris5342** [reply]
Update, I plugged in a router to the bypassed Starlink ethernet and read off:
IP Address 100.69.225.184
Netmask 255.192.0.0
Gateway 100.64.0.1

I plumbed these in to OMR and its working.

Still raises the problem of will these IPs change at some point ...

**Reply by redkite127** [reply]
Hi there, just stay in DHCP Client, no need of entering any value for address, netmask or gateway.

**Reply by Chris5342** [reply]
@redkite127 Oh. 
You are saying do not use bypass mode? 
Plug Starlink into a separate ethernet port (on my Pi)?
And use MacVLAN ? (whatever that is)
Thanks.

**Reply by Ysurac** [reply]
Personnaly I keep the Starlink router and use "192.168.1.1" as gateway. But I have the round dish.

**Reply by Chris5342** [reply]
@Ysurac but then you will have the Starlink DHCP which ia a no no?
Or does the round dish AUX port not have DHCP?
Sorry it's very confusing!

**Reply by Ysurac** [reply]
@Chris5342 it have DHCP, so you need to connect it to a dedicated port of the router or use VLAN. (Or use the bad solution I have, ignore the DHCP and use static address on OpenMTPCProuter config, the OpenMPTCProuter DHCP have a higher priority than Starlink DHCP and in most case clients computer get only IP via OpenMPTCProuter)

**Reply by kevin39** [reply]
Hello
I’ve the rounded dish, on my side I replaced the default router with a new one. This wan is bridged like other wan. 

I’m also able to play with static route to access to the Starlink status (192.168.100.1).

**Reply by Chris5342** [reply]
@Ysurac Gosh that was unexpected but helpful.

Starlink get 0/10 for not letting us access router setup ...

I was having big problems with using Gateway 100.64.0.1 (the interface would go dead for a minute or two every minute or two)
So I  put a spare (100M :( ) router (with DHCP turned off) between Starlink and OMR and it looks like it's finally working.

Gateway 100.64.0.1 had ping times of 30-100ms so is presumably elsewhere on the planet (or space) and maybe it doesn't like OMR traffic being sent to it and seems like a dumb plan...

@Ysurac is there a reason why you didn't go with this approach (or a dedicated port) ?
(I have never found DHCP clash to work at all)

Shall I summarise my experience and if so which section should I  put it in > Q&A? Show&Tell ?

Now I need to decide if I use a dedicated port via USB ethernet on my Pi or stick with a (better) router.
If I go with USB that will a whole load more pain I expect ...

**Reply by Chris5342** [reply]
@kevin39 Sounds similar to my approach.

Any reason you used a router and not a dedicated port ?

**Reply by Chris5342** [reply]
Wrote a guide here:
https://github.com/Ysurac/openmptcprouter/discussions/2353

Please let me know if I can make improvements / corrections.

**Reply by Network-Traditions** [reply]
Using StarLink v2 in bypass mode with ethernet adapter.  In MPTCP Router Settings wizard, add interface with DHCP.  OpenMPTCProuter will accept the CGNAT IP assignment and make the connection to your VPS.  Give it some time and it will eventually figure out the public IP address, properly connect and begin to carry traffic for the MP tunnel.

**Comment by lescdelac** [comment]
c'est une grosse blague ce routeur de starlink !!!! impossible de faire passer mon point d'accès !!! Il n'accepte uniquement qu'un autre routeur à la place en ne laissant passer que des complications !!! A te dégouter de starlink au final. La situation fait qu'on ne peut pas se permettre de n'avoir que starlink mais franchement ...
